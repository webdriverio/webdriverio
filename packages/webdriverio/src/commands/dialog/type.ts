/**
 * Returns dialog's type, can be one of `alert`, `beforeunload`, `confirm` or `prompt`.
 *
 * <example>
    :dialogType.js
    const type = await dialog.type();
 * </example>
 *
 * @alias dialog.type
 * @returns {string}  The type of the dialog
 */
// actual implementation is located in packages/webdriverio/src/dialog.ts
