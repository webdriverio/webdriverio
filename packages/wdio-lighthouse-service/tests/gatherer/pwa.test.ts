import { describe, expect, it, vi, beforeEach } from 'vitest'
import type { CDPSession } from 'puppeteer-core/lib/esm/puppeteer/common/Connection.js'
import type { Page } from 'puppeteer-core/lib/esm/puppeteer/api/Page.js'

import PWAGatherer from '../../src/gatherer/pwa.js'
import type { GathererDriver } from '../../src/types.js'

vi.mock('lighthouse/lighthouse-core/fraggle-rock/gather/session')
vi.mock('lighthouse/lighthouse-core/gather/gatherers/installability-errors')
vi.mock('lighthouse/lighthouse-core/gather/gatherers/web-app-manifest')
vi.mock('lighthouse/lighthouse-core/gather/gatherers/link-elements')
vi.mock('lighthouse/lighthouse-core/gather/gatherers/viewport-dimensions')
vi.mock('lighthouse/lighthouse-core/gather/driver/service-workers')

const pageMock = {
    on: vi.fn(),
    url: vi.fn().mockResolvedValue('http://json.org')
} as unknown as Page
const sessionMock = {
    on: vi.fn(),
    send: vi.fn(),
    _connection: { _transport: { _ws: { addEventListener: vi.fn() } } }
} as unknown as CDPSession

const driver = {
    evaluate: vi.fn()
} as any as GathererDriver

describe('PWAGatherer', () => {
    beforeEach(() => {
        vi.mocked(sessionMock.on).mockClear()
        vi.mocked(pageMock.on).mockClear()
    })

    it('initiates properly', () => {
        const pwaGatherer = new PWAGatherer(sessionMock, pageMock, driver)
        expect(vi.mocked(sessionMock.on).mock.calls.map((c) => c[0]))
            .toMatchSnapshot()
        pwaGatherer['_networkRecorder'].dispatch({ method: 'Network.requestWillBeSent', params: 'bar' })
        expect(pwaGatherer['_networkRecords']).toHaveLength(0)
        expect(pwaGatherer['_networkRecorder'].getRawRecords()).toHaveLength(1)
        vi.mocked(pageMock.on).mock.calls[0][1]({} as any)
        expect(pwaGatherer['_networkRecords']).toHaveLength(1)
        expect(pwaGatherer['_networkRecorder'].getRawRecords()).toHaveLength(0)
    })

    it('gatherData', async () => {
        const pwaGatherer = new PWAGatherer(sessionMock, pageMock, driver)
        pwaGatherer['_driver'].evaluate = vi.fn().mockResolvedValue({ some: 'result' })
        expect(await pwaGatherer.gatherData()).toMatchSnapshot()
    })
})
