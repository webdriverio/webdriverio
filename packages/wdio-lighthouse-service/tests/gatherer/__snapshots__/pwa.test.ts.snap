// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PWAGatherer > gatherData 1`] = `
{
  "InstallabilityErrors": "getInstallabilityErrorsResult",
  "LinkElements": "barfoo",
  "MetaElements": {
    "some": "result",
  },
  "ServiceWorker": {
    "registrations": 42,
    "versions": "1.2.1",
  },
  "URL": {
    "finalUrl": "http://json.org",
    "requestedUrl": "http://json.org",
  },
  "ViewportDimensions": "foobar",
  "WebAppManifest": "getWebAppManifestResult",
}
`;

exports[`PWAGatherer > initiates properly 1`] = `
[
  "Network.requestWillBeSent",
  "Network.requestServedFromCache",
  "Network.responseReceived",
  "Network.dataReceived",
  "Network.loadingFinished",
  "Network.loadingFailed",
  "Network.resourceChangedPriority",
]
`;
